rmw_opensplice_cpp/src/rmw_publisher.cpp

```diff
@@ -208,12 +222,57 @@ rmw_create_publisher(
       fprintf(stderr, "%s\n", check_delete_topic(status));
     }
   }
+  if (publisher_listener) {
+    RMW_TRY_DESTRUCTOR_FROM_WITHIN_FAILURE(
+      publisher_listener->~OpenSplicePublisherListener(), OpenSplicePublisherListener)
+    rmw_free(publisher_listener);
+  }
   if (publisher_info) {
+    if (publisher_info->listener) {
+      RMW_TRY_DESTRUCTOR_FROM_WITHIN_FAILURE(
+        publisher_info->listener->~OpenSplicePublisherListener(), OpenSplicePublisherListener)
+      rmw_free(publisher_info->listener);
+      publisher_info->listener = nullptr;
+    }
     rmw_free(publisher_info);
   }
+  if (listener_buf) {
+    rmw_free(listener_buf);
+  }
+
   return nullptr;
 }
 
+rmw_ret_t
+rmw_publisher_count_matched_subscriptions(
+  const rmw_publisher_t * publisher,
+  size_t * subscription_count)
+{
+  if (!publisher) {
+    RMW_SET_ERROR_MSG("publisher handle is null");
+    return RMW_RET_INVALID_ARGUMENT;
+  }
+
+  if (!subscription_count) {
+    RMW_SET_ERROR_MSG("subscription_count is null");
+    return RMW_RET_INVALID_ARGUMENT;
+  }
+
+  auto info = static_cast<OpenSpliceStaticPublisherInfo *>(publisher->data);
+  if (info == nullptr) {
```

Comment:
```suggestion
  if (!info) {
```

This is just inconsistent with the code above. An alternative to all of them (if you want to explicitly compare to `nullptr`) would be `nullptr == info` (following our "literal/constant on the left if possible" rule):

https://index.ros.org/doc/ros2/Developer-Guide/#c